dtonator/src/main/java/com/bizo/dtonator/GenerateEnum.java

```diff
@@ -44,6 +49,27 @@ private void addEnumValues() {
     }
   }
 
+  private void addEnumGetters() throws ClassNotFoundException {
+    final Class<?> clazz = Class.forName(dto.getDomainType());
+    for (final Method method : clazz.getDeclaredMethods()) {
+      if (method.getName().startsWith("get")) {
+        final GMethod getter = gc.getMethod("{}", method.getName()).returnType(method.getReturnType());
+        getter.body.line("switch (this) {");
+        for (final String value : dto.getEnumValues()) {
+          @SuppressWarnings({ "unchecked", "rawtypes" })
+          final Enum enumInstance = Enum.valueOf((Class<? extends Enum>) clazz, value);
+          try {
+            final Object returnValue = method.invoke(enumInstance);
+            getter.body.line("_ case {}: return {};", value, returnValue instanceof Integer ? returnValue : "\"" + returnValue.toString() + "\"");
+          } catch (IllegalAccessException | InvocationTargetException ignore) {
```

Comment:
I would be tempted to just catch all Exceptions, because when you call method.invoke, IAException and ITException are both reflection-related errors, but we could have the case where the reflection call "works" in that it executes the enum's code, but then the enum's code does something like "throw new NoDatabaseException()" or some other sort of business-logic/app-specific exception that makes sense in the context of their server-side/etc. environment.